README.md

```diff
@@ -1,4 +1,38 @@
 # Legobot
+## Legobot is an ongoing project; currently, refactoring is taking place. If you wish to contribute, please do so in `develop`.
+
+### Table of Contents
+
+1. Installation
+2. Connecting to IRC
+
+***
+
+## Introduction
+
+Legobot is a platform that aims to make interactive IRC bots less of a chore and more of a simple pleasure. Legobot itself is a skeleton framework to be built upon; Legobot provides the basic connectivity and ability to listen to message object (called threads) and respond to users based on the content of the message. Legobot is extensible via writing simple functions called "Legos", which can be easily added, edited, modified, and removed based on the need. The bot provides an easy-to-use "message" object when needed, which eliminated the need to parse raw IRC strings; this allows faster, more efficient processing of message contents, and quicker response time. Currently, IRC is the bot's main function; however, the bot is easily extensible to work with other messaging applications, or whatever you might find a need for. Legobot itself is not an IRC bot, it is an extensible, easy to use bot framework for quick, painless deployment of different forms of chat bots (or other if you so choose). 

```

Comment:
Message object isn't called a thread. Each Lego and Message is spun up on a thread, however. That's not exposed to the user or developer.